src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationMethod.java

```diff
@@ -0,0 +1,123 @@
+
+        // do normalization using actual score and min and max scores for corresponding sub query
+        for (CompoundTopDocs compoundQueryTopDocs : queryTopDocs) {
+            if (Objects.isNull(compoundQueryTopDocs)) {
+                continue;
+            }
+            List<TopDocs> topDocsPerSubQuery = compoundQueryTopDocs.getCompoundTopDocs();
+            for (int j = 0; j < topDocsPerSubQuery.size(); j++) {
+                TopDocs subQueryTopDoc = topDocsPerSubQuery.get(j);
+                for (ScoreDoc scoreDoc : subQueryTopDoc.scoreDocs) {
+                    scoreDoc.score = normalizeSingleScore(scoreDoc.score, minScoresPerSubquery[j], maxScoresPerSubquery[j]);
+                }
+            }
+        }
+    }
+
+    private float[] getMaxScores(List<CompoundTopDocs> queryTopDocs, int numOfSubqueries) {
+        float[] maxScores = new float[numOfSubqueries];
+        Arrays.fill(maxScores, Float.MIN_VALUE);
+        for (CompoundTopDocs compoundQueryTopDocs : queryTopDocs) {
+            if (Objects.isNull(compoundQueryTopDocs)) {
+                continue;
+            }
+            List<TopDocs> topDocsPerSubQuery = compoundQueryTopDocs.getCompoundTopDocs();
+            for (int j = 0; j < topDocsPerSubQuery.size(); j++) {
+                maxScores[j] = Math.max(
+                    maxScores[j],
+                    Arrays.stream(topDocsPerSubQuery.get(j).scoreDocs)
+                        .map(scoreDoc -> scoreDoc.score)
+                        .max(Float::compare)
+                        .orElse(Float.MIN_VALUE)
+                );
+            }
+        }
+        return maxScores;
+    }
+
+    private float[] getMinScores(List<CompoundTopDocs> queryTopDocs, int numOfScores) {
+        float[] minScores = new float[numOfScores];
+        Arrays.fill(minScores, Float.MAX_VALUE);
+        for (CompoundTopDocs compoundQueryTopDocs : queryTopDocs) {
+            if (Objects.isNull(compoundQueryTopDocs)) {
+                continue;
+            }
+            List<TopDocs> topDocsPerSubQuery = compoundQueryTopDocs.getCompoundTopDocs();
+            for (int j = 0; j < topDocsPerSubQuery.size(); j++) {
+                minScores[j] = Math.min(
+                    minScores[j],
+                    Arrays.stream(topDocsPerSubQuery.get(j).scoreDocs)
+                        .map(scoreDoc -> scoreDoc.score)
+                        .min(Float::compare)
+                        .orElse(Float.MAX_VALUE)
+                );
+            }
+        }
+        return minScores;
+    }
+
+    private float normalizeSingleScore(float score, float minScore, float maxScore) {
+        // edge case when there is only one score and min and max scores are same
+        if (Floats.compare(maxScore, minScore) == 0 && Floats.compare(maxScore, score) == 0) {
+            return SINGLE_RESULT_SCORE;
+        }
+        float normalizedScore = (score - minScore) / (maxScore - minScore);
```

Comment:
add a unit test when maxScore and minScore is same.

Reply:
ack, will add in next PR